lib/seattleflu/db/cli/command/reportable_conditions.py

```diff
@@ -0,0 +1,236 @@
+    """
+    Return sample organism and metadata from the reportable conditions view in
+    the *db* for matching presence-absence IDs from the given
+    *reportable_conditions*
+    """
+    ids = [ record.id for record in reportable_conditions ]
+
+    return db.fetch_all("""
+        select presence_absence_id as id, lineage, sample, barcode, site
+            from shipping.reportable_condition_v1
+            where presence_absence_id = ANY(%s)
+        """, (ids, ))
+
+
+def get_childrens_sites(db) -> List:
+    """Gets all sites from the warehouse whose name contains 'Childrens'"""
+
+    childrens_sites = db.fetch_all("""
+        select identifier
+            from warehouse.site
+            where identifier like '%Childrens%'
+        """, )
+    return [site.identifier for site in childrens_sites]
+
+
+def send_slack_post_request(record: NamedTupleCursor.Record, url: str) -> requests.PreparedRequest:
+    """
+    Sends a POST request to a channel-specific Slack webhook *url*. The payload
+    of this POST request is composed using Slack blocks. These blocks provide
+    structure for a nicely formatted message that contains a link to
+    Metabase plus relevant information from the given *record* from the
+    database. The message contains, by request, a machine-friendly Json document
+    containing minimal sample details.
+    """
+    data = {
+        "sample": record.barcode,
+        "site": record.site,
+        "condition": record.lineage
+    }
+
+    payload = {
+        "text": f":rotating_light: @channel {record.lineage} detected",
+        "blocks": [{
+            "type": "section",
+            "text": {
+                "type": "mrkdwn",
+                "text": dedent(f"""
+                :rotating_light: @channel {record.lineage} detected. \n
+                *<https://backoffice.seattleflu.org/metabase/question/55|Go to Metabase>*
+                """)
+            }
+        },
+        {
+            "type": "section",
+            "fields": [
+                {
+                    "type": "mrkdwn",
+                    "text": f"*Details:*\n```{json.dumps(data, sort_keys=True, indent=4)}```"
+                }
+            ]
+        }]
+    }
+
+    return requests.post(url, data=json.dumps(payload),
+                         headers={'Content-type': 'application/json'})
+
+
+def mark_processed(db, presence_absence_id: int, entry: {}) -> None:
+    LOG.debug(dedent(f"""
+    Marking reportable condition «{presence_absence_id}» as processed in the
+    presence_absence table"""))
+
+    data = {
+        "presence_absence_id": presence_absence_id,
+        "log_entry": Json({
+            **entry,
+            "revision": REVISION,
+            "timestamp": datetime.now(timezone.utc),
+        }),
+    }
+
+    with db.cursor() as cursor:
+        cursor.execute("""
+            update warehouse.presence_absence
+               set details = details || %(log_entry)s
```

Comment:
Yep, a nested object or array is what I was thinking.

Also, I would definitely guard the `set details` with a `coalesce` as you suggested. It will prevent us from overwriting existing details objects in the future if our loading changes.